addons/l10n_sa_edi/wizard/l10n_sa_edi_otp_wizard.py

```diff
@@ -0,0 +1,18 @@
+from odoo import fields, models, _
+from odoo.exceptions import UserError
+
+
+class RequestZATCAOtp(models.TransientModel):
+    _name = 'l10n_sa_edi.otp.wizard'
+    _description = 'Request ZATCA OTP'
+
+    l10n_sa_otp = fields.Char("OTP", copy=False, help="OTP required to get a CCSID. Can only be acquired through "
+                                                      "the Fatoora portal.")
+
+    def validate(self):
+        if not self.l10n_sa_otp:
+            raise UserError(_("You need to provide an OTP to be able to request a CCSID"))
+        journal_id = self.env['account.journal'].browse(self.env.context.get('active_id'))
```

Comment:
```suggestion
        journal = self.env['account.journal'].browse(self.env.context.get('active_id'))
```
Are you sure that active_id is still set here?  We could provide a journal_id field in this wizard.

Reply:
@jco-odoo Yes, I tried multiple scenarios to check if the active_id gets removed from the context, but nothing so far. Should we still create a journal_id field on the journal ?

Reply:
It is still clearer, yes.

Reply:
No update here? I agree with JCO better to use a field than rely on the context.